subsys/dfu/img_util/flash_block.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "fota/flash_block"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_FOTA_LEVEL
+#include <logging/sys_log.h>
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <flash.h>
+#include "boot/mcuboot_constrains.h"
+#include "img_util/flash_block.h"
+
+#if (CONFIG_IMG_BLOCK_BUF_SIZE % FLASH_ALIGN != 0)
+#error CONFIG_IMG_BLOCK_BUF_SIZE is not multiple of FLASH_ALIGN
+#endif
+
+static bool flash_verify(struct device *dev, off_t offset,
+			 u8_t *data, int len)
+{
+	int temp;
```

Comment:
Int seems like an unusual type for this. You probably want something a bit bigger than calling through the flash API for every 4 bytes. If anything, how about making it an explicit 32-bit value, or an array of uint8 values?